SS2-Project/Assets/Starstorm2/Modules/EntityStates/LampBoss/LampBossSpawn.cs

```diff
@@ -7,17 +7,66 @@ public class LampBossSpawn : GenericCharacterSpawnState
     {
         public static GameObject spawnVFX;
         public static GameObject spawnVFXblue;
-        public static string muzzleString;
+        private static string childString = "BodyMesh";
         private Transform muzzle;
-        private EffectData effectData;
+
+        private static float realDuration = 3.7f;
+        private static float printTime = 3.7f;
+        private static float printStartHeight = 9.5f;
+        private static float printMaxHeight = -5f;
+        private static float printStartBias = 6f;
+        private static float printMaxBias = 2f;
+
+        private Animator animator;
+        private AimAnimator aimAnimator;
+
+
         public override void OnEnter()
         {
+            base.damageStat = realDuration;
             base.OnEnter();
-            muzzle = GetModelChildLocator().FindChild(muzzleString);
+
+            // TODO: SPAWN ANIMATION!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+            animator = GetModelAnimator();
+            animator.enabled = false;
+            aimAnimator = GetAimAnimator();
+            aimAnimator.enabled = false;
+
             bool isBlue = GetModelTransform().GetComponentInChildren<ModelSkinController>().skins[characterBody.skinIndex].nameToken == "SS2_SKIN_LAMP_BLUE";
```

Comment:
Can we compare the SkinCatalog.SkinIndex of the skin instead of doing a string comparison?